tests/ir.at

```diff
@@ -0,0 +1,22 @@
+AT_BANNER([intermediate representation unit tests])
```

Comment:
m4 syntax takes some getting used to, particularly the automake dialect of it.

This still defines the same macro mulitple times with different names (just once in each file) -- might it make more sense to call it `P4TEST_TEST_RUN` or (something less verbose) and put it in the top level file (testsuite.at?)  That way each sub .at file is just a list of tests.